androidbound-support/src/main/java/solutions/alterego/androidbound/support/android/ui/BindableCompatEditText.java

```diff
@@ -0,0 +1,214 @@
+
+public class BindableCompatEditText extends AppCompatEditText implements INotifyPropertyChanged {
+
+    protected BindableViewDelegate mDelegate;
+
+    public BindableCompatEditText(Context context) {
+        this(context, null);
+    }
+
+    public BindableCompatEditText(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        mDelegate = createDelegate(this);
+    }
+
+    public BindableCompatEditText(Context context, AttributeSet attrs, int defStyle) {
+        super(context, attrs, defStyle);
+        mDelegate = createDelegate(this);
+    }
+
+    private final TextWatcher textWatcher = new TextWatcher() {
+        @Override
+        public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+
+        }
+
+        @Override
+        public void onTextChanged(CharSequence s, int start, int before, int count) {
+
+        }
+
+        @Override
+        public void afterTextChanged(Editable s) {
+            mDelegate.notifyPropertyChanged("TextString");
+        }
+    };
+
+    @Override
+    protected void onAttachedToWindow() {
+        super.onAttachedToWindow();
+        addTextChangedListener(textWatcher);
+    }
+
+    @Override
+    protected void onDetachedFromWindow() {
+        super.onDetachedFromWindow();
+        removeTextChangedListener(textWatcher);
+    }
+
+    /****** beginning of the delegated methods, to be copy/pasted in every bindable view ******/
+
+    protected BindableViewDelegate createDelegate(View view) {
+        return new BindableViewDelegate(view);
+    }
+
+    public ICommand getClick() {
+        return mDelegate.getClick();
+    }
+
+    public void setClick(ICommand value) {
+        mDelegate.setClick(value);
+    }
+
+    public ICommand getLongClick() {
+        return mDelegate.getLongClick();
+    }
+
+    public void setLongClick(ICommand value) {
+        mDelegate.setLongClick(value);
+    }
+
+    public int getBackgroundColor() {
+        return mDelegate.getBackgroundColor();
+    }
+
+    public void setBackgroundColor(int color) {
+        mDelegate.setBackgroundColor(color);
+        super.setBackgroundColor(color);
+    }
+
+    public StateListDrawable getBackgroundDrawableState() {
+        return mDelegate.getBackgroundDrawableState();
+    }
+
+    public void setBackgroundDrawableState(StateListDrawable colors) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
+            super.setBackground(colors);
+            mDelegate.setBackgroundDrawableState(colors);
+        }
+    }
+
+    public int getBackgroundResource() {
+        return 0;
+    }
+
+    public void setBackgroundResource(int res) {
+        super.setBackgroundResource(res);
+    }
+
+    public int getBackgroundDrawable() {
+        return 0;
+    }
+
+    public void setBackgroundDrawable(Drawable res) {
+        super.setBackgroundDrawable(res);
+    }
+
+    public Typeface getTypeface() {
```

Comment:
get and set typeface are just noise. remove them please

Reply:
you're right, removed them.